modules/docs/mdx/7.0-MIGRATION-GUIDE.mdx

```diff
@@ -538,6 +539,242 @@ main `react` package.
 - `Flex` was promoted from `@workday/canvas-kit-labs/layout` to `@workday/canvas-kit-react/layout`
 - `Stack` was promoted from `@workday/canvas-kit-labs/layout` to `@workday/canvas-kit-react/layout`
 
+## Model Changes
+
+### Guards and Callbacks
+
+We changed the signature of model event guards and callbacks. In v6, the parameters were in an
+object. This has a less than ideal developer experience as intellisense isn't engaged immediately
```

Comment:
When you say "intellisense isn't engaged immediately" are we describing that IntelliSense slower with an object parameter?

I will personally complain that with the object, it was one extra level of destructuring to get what I wanted. I agree with making prevState the second parameter too. This makes sense.

Reply:
Yes. It is an additional level. The parameter given to callbacks and guards was called `event`, and the properties were `data`, and `state` for guards  and `prevState` for callbacks. This often lead to odd paths for events like `show` and `hide` where the parameter for the DOM event was also called `event`: `event.data.event`.